### ISSUE TITLE
Make sure at most once `Updating Git Machete status` background task runs at any given moment

### ISSUE THREAD

**Issue by PawelLipski** [VirtusLab/git-machete-intellij-plugin#1385]
Might be hard to ensure... still, would be a nice optimization on large repositories.

**Comment by mkondratek** [comment]
good point! I was about to open the same issue 😆 I have already encountered the situation with +1 tasks running 😞

**Comment by PawelLipski** [comment]
At least two approaches to be considered:
1. not allow enqueuing a new task if any is running
  - plus: old tasks can finish faster, won't get cancelled or get any competition for disk/CPU from newer tasks
  - minus: old tasks might not reflect the most state of the repo
2. cancelling the existing task if new one is scheduled
  - plus: new tasks will reflect the latest state
  - minus: might take long if an almost-ready task is always getting cancelled in favor of a new task

**Comment by PawelLipski** [comment]
Might no longer be important once #1395 is released... to be checked on our large repository (650k commits). The early results look promising.

**Comment by PawelLipski** [comment]
Doesn't seem crucial anymore!

**Comment by PawelLipski** [comment]
Nopes, it still seems after all 😅 I've just got 6 tasks running at the same time on my large commercial repo.

**Comment by PawelLipski** [comment]
One thing to consider: sometimes the earlier-triggered background task ends up later, which causes the state displayed to the user to be out of date :/
